engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableAggregationTest.java

```diff
@@ -3824,4 +3834,97 @@ public void testPreserveEmptyNoKey() {
         TestCase.assertEquals(1, aggregated.size());
         assertTableEquals(expectedEmpty, aggregated);
     }
+
+    @Test
+    public void testMultiPartitionSymbolTableBy() throws IOException {
+        final File testRootFile = Files.createTempDirectory(QueryTableAggregationTest.class.getName()).toFile();
+        try {
+            final Table t1 = new InMemoryTable(
+                    new String[] {"StringKeys", "GroupedInts"},
+                    new Object[] {
+                            new String[] {"key1", "key2", "key1", "key2", "key1", "key2", "key1", "key2", "key1"},
+                            new short[] {1, 1, 2, 2, 2, 3, 3, 3, 3}
+                    });
+            final Table t1_asc = t1.sort("StringKeys");
+            final Table t1_desc = t1.sortDescending("StringKeys");
```

Comment:
I would like to see a case where the sets of defined symbols are not identical. I would leave the overall introduction order aligned with the names, just to keep things simple in a debugger. I would also like to see `null` symbols exercised.

Suggested:
{key1, key1,  key2, key2, key3, key2, key4, key2, key1}
{key4, null, key3, null, key1, null, null}
{key5, key4, key2, key5, key6}
{null, key6, key6, key6, key6, key7, null}

Be sure to run this test with coverage to make sure we're hitting the intended cases.